components/Pagination/src/Pagination.tsx

```diff
@@ -0,0 +1,47 @@
+import React, { useEffect } from 'react';
+import clsx from 'clsx';
+import { PaginationPrevious } from './PaginationPrevious';
+import { PaginationNext } from './PaginationNext';
+import { PaginationList } from './PaginationList';
+import { PaginationLink } from './PaginationLink';
+
+interface Props {
+  index?: number;
+  indexLimit: number;
+  onChange?: (index: number) => number;
+  className?: string;
+}
+
+export const Pagination = ({ index = 0, indexLimit, onChange, className }: Props) => {
+  const [currentIndex, setCurrentIndex] = React.useState(index);
+  const pages = Array.from({ length: indexLimit + 1 }, (_, i) => i + 1);
+
+  useEffect(() => {
+    setCurrentIndex(index);
+  }, [index]);
+
+  const changePage = (index: number) => {
+    if (index < 0 || index > indexLimit) return;
+    setCurrentIndex(index);
+    onChange?.(index);
+  };
+
+  return (
+    <nav className={clsx('denhaag-pagination', className)}>
```

Comment:
Ik zou de nav ook in een los React component zetten voor de flexibiliteit, ook al is dat lastig qua naamgeving. Stel iemand wilt de Pagination gebruiken maar met een net iets andere structuur of werking, dan kan diegene gewoon de PaginationNav gebruiken.